prov/efa/src/efa_device.c

```diff
@@ -503,3 +508,31 @@ int efa_device_get_pci_attr(struct efa_device *device,
 }
 
 #endif // _WIN32
+
+#if EFA_UNIT_TEST
+
+#include "efa_unit_tests.h"
+
+/*
+ * test the error handling path of efa_device_construct()
+ */
+void test_efa_device_construct_error_handling()
+{
+	int ibv_err = 4242;
+	struct ibv_device **ibv_device_list;
+	struct efa_device efa_device = {0};
+
+	ibv_device_list = ibv_get_device_list(&g_device_cnt);
+	if (ibv_device_list == NULL) {
+		skip();
+		return;
+	}
+
+	will_return(__wrap_efadv_query_device, ibv_err);
+	efa_device_construct(&efa_device, 0, ibv_device_list[0]);
+
+	/* when error happend, efa_device.ibv_ctx should be NULL */
+	assert_null(efa_device.ibv_ctx);
```

Comment:
why not we check `efa_device->rdm_info` and `efa_device->dgram_info`, which are not set to NULL in the err handling path before the bug fix?

Reply:
make sense. added in new revision.